### PR TITLE
Docker manifest (schemaVersion 1) to package version

### PR DESCRIPTION
**Pull request by mgrand**
This is the first installment of extracting a packageVersion metadata record from a docker manifest. 

This produces unsigned metadata. This is temporary. It will be moved outside the node where it can be securely signed.

### CODE REVIEW

File: pyrsia_node/src/docker/v2/handlers/manifests.rs [hunk 1]
```diff
@@ -212,17 +234,196 @@ fn store_manifest_in_artifact_manager(bytes: &Bytes) -> anyhow::Result<(HashAlgo
...
+            .context("invalid fslayer")?
+            .get("blobSum")
+            .context("missing blobSum")?
+            .as_str()
+            .context("invalid blobSum")?;
+        if !hex_digest.starts_with("sha256:") {
+            return Err(anyhow!("Only sha256 digests are supported: {}", hex_digest));
+        }
+        let digest = hex::decode(&hex_digest["sha256:".len()..])?;
+        artifacts.push(Artifact::new(
```
Comment (efrisch) [inline pyrsia_node/src/docker/v2/handlers/manifests.rs:311]: This is ugly. Can we consider using a builder for constructing Artifacts?

File: pyrsia_node/src/docker/v2/handlers/manifests.rs [hunk 2]
```diff
@@ -212,17 +234,196 @@ fn store_manifest_in_artifact_manager(bytes: &Bytes) -> anyhow::Result<(HashAlgo
...
+            None,
+            None,
+            None,
+            None,
+            None,
+            Map::new(),
+            None,
+        ))
+    }
+    Ok(PackageVersion::new(
```
Comment (efrisch) [inline pyrsia_node/src/docker/v2/handlers/manifests.rs:323]: And a builder for PackageVersion(s).

File: pyrsia_node/src/docker/v2/handlers/manifests.rs [hunk 3]
```diff
@@ -212,17 +234,196 @@ fn store_manifest_in_artifact_manager(bytes: &Bytes) -> anyhow::Result<(HashAlgo
...
+    ))
+}
+
+fn package_version_from_schema2(
+    _json_object: &Map<String, Value>,
+    _json_string: &str,
+    _docker_name: &str,
+    _docker_reference: &str,
+) -> Result<PackageVersion, anyhow::Error> {
+    todo!()
```
Comment (efrisch) [inline pyrsia_node/src/docker/v2/handlers/manifests.rs:351]: That's new to me. Nice.

File: pyrsia_node/src/node_manager/model/artifact.rs [hunk 4]
```diff
@@ -35,11 +35,74 @@ pub struct Artifact {
...
     /// The mime type of the artifact
     mime_type: Option<String>,
     /// Attributes of an artifact that don't fit into one of this struct's fields can go in here as JSON
     metadata: Map<String, Value>,
     /// The URL of the source of the artifact
     source_url: Option<String>,
 }
+
+impl Artifact {
+    #[allow(clippy::too_many_arguments)]
```
Comment (efrisch) [inline pyrsia_node/src/node_manager/model/artifact.rs:48]: Poor clippy.

**Review by efrisch — APPROVED** [review]
👍

**Review by chb0github — APPROVED** [review]

File: pyrsia_node/src/docker/v2/handlers/manifests.rs (hunk 1 above)
Comment (mgrand) [reply pyrsia_node/src/docker/v2/handlers/manifests.rs:311]: Ok. There is a derive_builder crate that provides a 
`#[derive(Builder)]`
macro.  I will use that for Artifacts. 
For the structs that we are using the 
`#[signed_struct]`
macro, it is a little bit more complicated. I will write an issue to add that later.

File: pyrsia_node/src/docker/v2/handlers/manifests.rs (hunk 2 above)
Comment (mgrand) [reply pyrsia_node/src/docker/v2/handlers/manifests.rs:323]: Yes. That is one of the structs that uses #[signed_struct]

**Pull request merged**